torch/csrc/autograd/python_variable.cpp

```diff
@@ -231,6 +231,22 @@ int THPVariable_set_data(THPVariable *self, PyObject *data)
   END_HANDLE_TH_ERRORS_RET(-1)
 }
 
+static PyObject * THPVariable_get_int_repr(THPVariable *self)
```

Comment:
hm, so what should int_repr() return? One option is a tensor with the view on the same underlying DataPtr. I don't think we can return Storage because dtype wouldn't match.

Reply:
can we return a numpy object?

Reply:
it's a bit weird to get back numpy, I think getting tensor is pretty logical.

The annoying part is how to get the view on the same storage but with different data type (we now enforce data type on storage). One hacky option I can imagine is to create a new Storage and Tensor where the DataPtr underneath is a pointer to the same dataptr storage but captures original storage in its "data_ptr::context" (to make sure that it stays around). I guess that would work and even would give you a mutable view (which would go bad though if user reallocates the original tensor, but it's ok). @gchanan - what do you think?